lib/stream/leo-stream.js

```diff
@@ -1275,207 +1278,208 @@ module.exports = function(configure) {
-												done(err);
-											} else {
+												logger.debug("skipping gzip");
 												done();
 											}
-										});
-									} else {
-										done();
-									}
-								} else if (!item.gzip || item.gzip == true) {
-									item.eid = item.kinesis_number;
-									delete item.kinesis_number;
-									// v1 getEvents already unzipped the payload
-									if (item.gzip && leoEvent.v >= 2) {
-										try {
-											item.payload = JSON.parse(zlib.gunzipSync(item.payload).toString());
-										} catch (e) {
-											item.payload = {};
-										}
-									} else if (typeof item.payload === "string") {
-										item.payload = JSON.parse(item.payload);
-									}
-
-									if (!isPassDestroyed && item.eid.localeCompare(start) > 0 && totalCount < opts.limit && totalSize < opts.size) { //Then it is greater than the event they were looking at
-										totalCount++;
-										pass.throttledWrite(item, done);
-									} else { //otherwise it means we had a number in the middle of a file
-										logger.debug("skipping gzip");
-										done();
-									}
-								} else if (item.gzip) {
-									var gzip = zlib.createGunzip();
-									gzip.setEncoding('utf8');
-									gzip.on('error', function(err) {
-										logger.debug("Skipping gzip");
-									});
-									pump(gzip, split(JSON.parse), ls.write((e, done) => {
-										e.eid = createEId(e.eid);
-										if (!isPassDestroyed && e.eid.localeCompare(start) > 0 && totalCount < opts.limit && totalSize < opts.size) { //Then it is greater than the event they were looking at
-											totalCount++;
-											pass.throttledWrite(e, done);
-										} else { //otherwise it means we had a number in the middle of a file
-											logger.debug("skipping gzipped");
-											done();
+										} else if (item.gzip) {
+											var gzip = zlib.createGunzip();
+											gzip.setEncoding('utf8');
+											gzip.on('error', function(err) {
+												logger.debug("Skipping gzip");
+											});
+											pump(gzip, split(JSON.parse), ls.write((e, done) => {
+												e.eid = createEId(e.eid);
+												if (!isPassDestroyed && e.eid.localeCompare(start) > 0 && totalCount < opts.limit && totalSize < opts.size) { //Then it is greater than the event they were looking at
+													totalCount++;
+													pass.throttledWrite(e, done);
+												} else { //otherwise it means we had a number in the middle of a file
+													logger.debug("skipping gzipped");
+													done();
+												}
+											}), (err) => {
+												if (err) {
+													logger.error(err);
+													done(err);
+												} else {
+													logger.debug("gzipped event read finished", item.records, totalCount);
+													done();
+												}
+											});
+											gzip.end(item.gzip);
 										}
-									}), (err) => {
+									},
+									function(err) {
+										start = items[items.length - 1].end + " ";
```

Comment:
It would be helpful if there was a comment here stating where start is used. I see it was defined outside the scope of this function, so this function is just updating it, but then where is it used? Maybe I'm full of it, but it's hard to follow in a CR.